activerecord/lib/active_record/associations/preloader/branch.rb

```diff
@@ -0,0 +1,113 @@
+# frozen_string_literal: true
+
+module ActiveRecord
+  module Associations
+    class Preloader
+      class Branch #:nodoc:
+        attr_reader :association, :children, :parent
+        attr_reader :scope, :associate_by_default
+        attr_writer :preloaded_records
+
+        def initialize(association:, children:, parent:, associate_by_default:, scope:)
+          @association = association
+          @parent = parent
+          @scope = scope
+          @associate_by_default = associate_by_default
+
+          @children = build_children(children)
+        end
+
+        def children_hash
+          @children.map do |child|
+            [child.association, child.children_hash]
+          end.to_h
```

Comment:
Nitpick:
```suggestion
          @children.to_h { |child| [child.association, child.children_hash] }
```

Reply:
oops. I think I can delete this entire method